### PR TITLE
Introduce minimum test coverage thresholds, coveralls.io for threshold reporting and repo badge

### LINKED ISSUE #113: Introduce minimum test coverage for Jest, coveralls.io
_To be completed after #106_

As we migrate our tests away from JavaScript and Jasmine towards TypeScript and Jest, we should begin to enforce minimum test coverage. We should also integrate and badge the repository with [coveralls.io](https://www.coveralls.io) so that we can monitor the amount of code coverage.

- Add and progressively raise the minimum test coverage percentage for Jest
- Introduce coveralls.io, and set a minimum threshold failure of 1.0% in coveralls.io settings to allow for code refactorings that may decrease test coverage slightly.
- Add coveralls.io badge to repository readme

### PR DESCRIPTION
**Pull request by liangyuanruo**
## Problem

We want to get better test coverage reports and enforce a minimum threshold of coverage as we continue to refactor the codebase.

Closes #113

## Solution

Introduce coveralls.io which integrates with Jest for test reporting, and configure Jest to enforce minimum test coverage (currently 12%). This figure should be progressively raised as more tests get ported from Jasmine to Jest.

**Features**:

- Test coverage reports available on [coveralls.io](https://coveralls.io/github/opengovsg/FormSG) and locally.

**Improvements**:

- Enforces test coverage to 12%

## Screenshots

image

## Deploy Notes

_Notes regarding deployment of the contained body of work. These should note any
new dependencies, new scripts, etc._

**New dev dependencies**:

- `coveralls` : Reporting tool for uploading Jest's coverage report to coveralls.io

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -8,4 +8,11 @@ module.exports = {
     // Needed to use @shelf/jest-mongodb preset.
     ...tsJestPreset.transform,
   },
+  collectCoverageFrom: ['./src/**/*.{ts,js}'],
```
Comment (karrui) [inline jest.config.js:11]: Should we only collect coverage for `ts` files, since jest only works for ts files currently?

File: jest.config.js [hunk 2]
```diff
@@ -8,4 +8,11 @@ module.exports = {
...
+  coveragePathIgnorePatterns: ['./node_modules/', './tests'],
+  coverageThreshold: {
+    global: {
+      statements: 12, // Increase this percentage as test coverage improves
```
Comment (karrui) [inline jest.config.js:15]: I think the current coverage has jumped to above 14%, might as well change it now :P

**Review by karrui — APPROVED** [review]
lgtm, with some comments up to you to change

**Comment by liangyuanruo** [comment]
@karrui Jest flags zero test coverage if we remove JS files. will revert to previous implementation

**Pull request merged**